src/fu-security-attr.c

```diff
@@ -203,6 +203,598 @@ fu_security_attr_get_name(FwupdSecurityAttr *attr)
+		} else {
+			/* TRANSLATORS: longer failure message */
+			g_ptr_array_add(strs, _("System firmware verification has been disabled."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_CET_ENABLED) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer success message (unlikely to see) */
+					_("The processor could detect and ignore more attack "
+					  "attempts from malicious software."));
+		} else {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer failure message */
+					_("The processor cannot detect and ignore more attack "
+					  "attempts from malicious software."));
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_CET_ACTIVE) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer success message (unlikely to see) */
+					_("The processor will detect and ignore more attack "
+					  "attempts from malicious software."));
+		} else {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer failure message */
+					_("The processor will not detect and ignore more attack "
+					  "attempts from malicious software."));
+			check_bios_settings = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_SMAP) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer success message (unlikely to see) */
+					_("The processor can detect and ignore simple attack "
+					  "attempts from malicious software."));
+		} else {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer failure message */
+					_("The processor cannot detect and ignore simple attack "
+					  "attempts from malicious software."));
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_ENCRYPTED_RAM) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer success message (unlikely to see) */
+					_("Memory is being encrypted and cannot be read if removed "
+					  "by an attacker."));
+		} else {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer failure message */
+					_("Memory is not encrypted and may be readable if removed "
+					  "by an attacker."));
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_IOMMU) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(
+			    strs,
+			    /* TRANSLATORS: longer success message (unlikely to see) */
+			    _("Malicious peripherals cannot write to main system memory."));
+		} else {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer failure message */
+					_("Some malicious peripherals could potentially write to "
+					  "main system memory."));
+			check_bios_settings = TRUE;
```

Comment:
I'm not sure I see the value of IOMMU and pre-boot IOMMU being a separate thing.  If pre-boot IOMMU is enabled the Linux kernel automatically turns on IOMMU.  The only way to avoid it is custom kernel command line then in Linux.

Reply:
I don't disagree, and I don't think it makes sense to show any status IOMMU if the Pre-boot IOMMU isn't enabled. Maybe we should use the obsoletes feature for that?